util/subnetworkid/subnetworkid_test.go

```diff
@@ -0,0 +1,443 @@
+		t.Errorf("String: wrong SubnetworkID string - got %v, want %v",
+			SubnetworkIDStr, wantStr)
+	}
+}
+
+func TestSubnetworkIDesStrings(t *testing.T) {
+	first := SubnetworkID{
+		0x06, 0xe5, 0x33, 0xfd, 0x1a,
+		0xda, 0x86, 0x39, 0x1f, 0x3f,
+		0x6c, 0x34, 0x32, 0x04, 0xb0,
+		0xd2, 0x78, 0xd4, 0xaa, 0xec,
+	}
+	firstStr := "ecaad478d2b00432346c3f1f3986da1afd33e506"
+
+	second := SubnetworkID{}
+	secondStr := "0000000000000000000000000000000000000000"
+
+	tests := []struct {
+		name            string
+		SubnetworkIDes  []SubnetworkID
+		expectedStrings []string
+	}{
+		{"empty", []SubnetworkID{}, []string{}},
+		{"two SubnetworkIDes", []SubnetworkID{first, second}, []string{firstStr, secondStr}},
+		{"two SubnetworkIDes inversed", []SubnetworkID{second, first}, []string{secondStr, firstStr}},
+	}
+
+	for _, test := range tests {
+		strings := Strings(test.SubnetworkIDes)
+		if !reflect.DeepEqual(strings, test.expectedStrings) {
+			t.Errorf("SubnetworkIDesStrings: %s: expected: %v, got: %v",
+				test.name, test.expectedStrings, strings)
+		}
+	}
+}
+
+// TestNewFromStr executes tests against the NewFromStr function.
+func TestNewFromStr(t *testing.T) {
+	tests := []struct {
+		in   string
+		want SubnetworkID
+		err  error
+	}{
+
+		// Empty string.
+		{
+			"",
+			SubnetworkID{},
+			nil,
+		},
+
+		// Single digit SubnetworkID.
+		{
+			"1",
+			SubnetworkID([IDLength]byte{
+				0x01, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x00, 0x00, 0x00, 0x00,
+			}),
+			nil,
+		},
+
+		// Block 203707 with stripped leading zeros.
+		{
+			"a60840790ba1d475d01367e7c723da941069e9dc",
+			SubnetworkID([IDLength]byte{
+				0xdc, 0xe9, 0x69, 0x10, 0x94,
+				0xda, 0x23, 0xc7, 0xe7, 0x67,
+				0x13, 0xd0, 0x75, 0xd4, 0xa1,
+				0x0b, 0x79, 0x40, 0x08, 0xa6,
+			}),
+			nil,
+		},
+
+		// SubnetworkID string that is too long.
+		{
+			"01234567890123456789012345678901234567890123456789012345678912345",
+			SubnetworkID{},
+			ErrIDStrSize,
+		},
+
+		// SubnetworkID string that is contains non-hex chars.
+		{
+			"abcdefg",
+			SubnetworkID{},
+			hex.InvalidByteError('g'),
+		},
+	}
+
+	unexpectedErrStr := "NewFromStr #%d failed to detect expected error - got: %v want: %v"
+	unexpectedResultStr := "NewFromStr #%d got: %v want: %v"
+	t.Logf("Running %d tests", len(tests))
+	for i, test := range tests {
+		result, err := NewFromStr(test.in)
+		if !errors.Is(err, test.err) {
+			t.Errorf(unexpectedErrStr, i, err, test.err)
+			continue
+		} else if err != nil {
+			// Got expected error. Move on to the next test.
+			continue
+		}
+		if !test.want.IsEqual(result) {
+			t.Errorf(unexpectedResultStr, i, result, &test.want)
+			continue
+		}
+	}
+}
+
+// TestAreEqual executes tests against the AreEqual function.
+func TestAreEqual(t *testing.T) {
+	SubnetworkID0, _ := NewFromStr("0000000000000000000000000000000000000000")
+	SubnetworkID1, _ := NewFromStr("1111111111111111111111111111111111111111")
```

Comment:
In many contexts SubnetworkID 1 (a.k.a. coinbase) is 0000000000000000000000000000000000000001.
So I'd rather, to avoid confusion, if the value was the above (here and in all subnetworkIDN's), or you rename to something like `SubnetworkIDAll1s`